### PR TITLE
Consider only images starting with `dataproc-<major>-<minor>` as base

### PR DESCRIPTION
**Pull request by vinayakumarb**
While resolving base images using image version, consider only images starting with `dataproc-` as base images.

There may be other custom images in the cloud-dataproc project itself, but they are not the original release images.

### CODE REVIEW

**Review by medb — APPROVED** [review]

File: custom_image_utils/args_inferer.py [hunk 1]
```diff
@@ -165,9 +165,13 @@ def _get_dataproc_image_path_by_version(version):
     if stdout:
       # in case there are multiple images
       parsed_lines = stdout.decode('utf-8').strip().split('\n')
+      expected_prefix = "dataproc-{}-{}".format(parsed_version[0], parsed_version[1])
       for line in parsed_lines:
         parsed_image = line.split(",")
         if len(parsed_image) == 2:
+          if not parsed_image.startswith(expected_prefix):
+            # Not a regular dataproc release image. Maybe a custom image with same label.
```
Comment (functicons) [inline custom_image_utils/args_inferer.py:173]: Add a log for it.

Comment (functicons) [reply custom_image_utils/args_inferer.py:173]: Can you also add a check that if more than one valid images are found, fail the build with an error.

**Review by functicons — CHANGES_REQUESTED** [review]

File: custom_image_utils/args_inferer.py (hunk 1 above)
Comment (vinayakumarb) [reply custom_image_utils/args_inferer.py:184]: Done.

File: custom_image_utils/args_inferer.py [hunk 2]
```diff
@@ -129,50 +129,79 @@ def _get_dataproc_image_path_by_version(version):
...
+      expected_prefix = "dataproc-{}-{}".format(major_version, minor_version)
+      _LOG.info("Filtering images : %s", expected_prefix)
+      image_versions=[]
+      all_images_for_version = {}
       for line in parsed_lines:
         parsed_image = line.split(",")
         if len(parsed_image) == 2:
-          return (_IMAGE_PATH.format("cloud-dataproc",
-                                     parsed_image[0]), parsed_image[1])
+          if not parsed_image[0].startswith(expected_prefix):
```
[the remark this answers is not in the document]
Comment (vinayakumarb) [reply custom_image_utils/args_inferer.py]: Done.

File: custom_image_utils/args_inferer.py [hunk 3]
```diff
@@ -129,50 +129,79 @@ def _get_dataproc_image_path_by_version(version):
...
+          if parsed_image[1] not in all_images_for_version:
+            all_images_for_version[parsed_image[1]] = [_IMAGE_PATH.format("cloud-dataproc", parsed_image[0])]
+            image_versions.append(parsed_image[1])
+          else:
+            all_images_for_version[parsed_image[1]].append(_IMAGE_PATH.format("cloud-dataproc", parsed_image[0]))
+
+      _LOG.info("All Images : %s", all_images_for_version)
+      _LOG.info("All Image-Versions : %s", image_versions)
+
+      for image_version in image_versions:
```
[the remark this answers is not in the document]
Comment (vinayakumarb) [reply custom_image_utils/args_inferer.py]: I have added validation to only the latest image which supposed to be used as a base image if validation passes. As suggested above checking for all versions is not required, even with flags, in this script's scope.

Validation to ensure one release image for a single version label may be ensured by a separate prober internally or during the release. So this validation is out of scope of this script and not added.

**Pull request merged**